src/components/screens/on-call-calendar-screen.tsx

```diff
@@ -102,8 +102,19 @@ const OnCallCalendarScreen = () => {
       week: weekNumber,
       paid: !paid
     };
-    await onCallApi.updatePaid({ onCallPaid: updateParameters });
-    refreshOnCallData();
+    try {
+      await onCallApi.updatePaid({ onCallPaid: updateParameters });
+      // Locally update paid status
+      setOnCallData(prev =>
+        prev.map(item =>
+          item.year === year && item.week === weekNumber
+            ? { ...item, paid: !paid }
+            : item
+        )
+      );
+    } catch (error) {
+      setError(`${error}`);
```

Comment:
In the setError a localized string to give more context e.g. "Error while updating on call paid status"